dask/dataframe/core.py

```diff
@@ -7696,6 +7696,10 @@ def repartition(df, divisions=None, force=False):
     >>> ddf = dd.repartition(df, [0, 5, 10, 20])  # doctest: +SKIP
     """
 
+    # no-op fastpath for when we already have matching divisions
+    if is_dask_collection(df) and df.divisions == divisions:
```

Comment:
This test is failing if divisions is a `list`, it only works with `tuples`

Reply:
It's an issue because if we concat a few Dask DataFrames with identical divisions then it will still repartition every dds:
`dask.dataframe.multi.concat()` calls `align_partitions()` that calls `df.repartition()` with a `list` divisions

Reply:
Shall I open a separate issue?

Reply:
Separate issue would be good, thanks @epizut